op-challenger/game/fault/register.go

```diff
@@ -84,8 +84,13 @@ func RegisterGameTypes(
 		}
 	}
 	if cfg.TraceTypeEnabled(faultTypes.TraceTypeAsterisc) {
-		if err := registerAsterisc(faultTypes.AsteriscGameType, registry, oracles, ctx, systemClock, l1Clock, logger, m, cfg, syncValidator, rollupClient, txSender, gameFactory, caller, l2Client, l1HeaderSource, selective, claimants); err != nil {
-			return nil, fmt.Errorf("failed to register asterisc game type: %w", err)
+		if err := registerAsterisc(faultTypes.AsteriscGameType, registry, oracles, ctx, systemClock, l1Clock, logger, m, cfg, syncValidator, rollupClient, txSender, gameFactory, caller, l2Client, l1HeaderSource, selective, claimants, faultTypes.ServerTypeOpProgram); err != nil {
+			return nil, fmt.Errorf("failed to register op-program-asterisc game type: %w", err)
+		}
+	}
+	if cfg.TraceTypeEnabled(faultTypes.TraceTypeAsterisc) {
+		if err := registerAsterisc(faultTypes.KonaAsteriscGameType, registry, oracles, ctx, systemClock, l1Clock, logger, m, cfg, syncValidator, rollupClient, txSender, gameFactory, caller, l2Client, l1HeaderSource, selective, claimants, faultTypes.ServerTypeKona); err != nil {
```

Comment:
hmm, I think we might want to be able to enable/disable each game type separately in challenger - its quite possible we'll have installations with `asterisc` and `op-program` available but no `kona`. With this setup once you enable `asterisc/op-program` it will also try to play `asterisc/kona` games and lose when it eventually finds it doesn't have a `kona` to execute.

Reply:
Why would it not have kona if the game type isn't actually on chain? Same thing with asterisc + op-program, which is already in there and not available in the DGF on any canonical chain.

Reply:
Think I got what you're saying here - added the ability to define supported server types as well as trace types. This'll filter out asterisc + kona in the register. This still isn't optimal, but a further refactor is out of scope of this PR.